shared/dataflow/codeql/dataflow/internal/DataFlowImpl.qll

```diff
@@ -171,6 +179,56 @@ module MakeImpl<LocationSig Location, InputSig<Location> Lang> {
     }
   }
 
+  /**
+   * Constructs a data flow computation given a full input configuration, and
+   * an initial stage 1 pruning with merging of overlay and base results.
+   */
+  module OverlayImpl<FullStateConfigSig Config, Stage1Output<Config::FlowState> Stage1> {
+    module Base = Impl<Config, Stage1>;
+
+    import Base
+
+    /**
+     * Holds if data can flow from `source` to `sink`.
+     *
+     * This is a local predicate that only has results local to the overlay/base database.
+     */
+    predicate flowLocal(Node source, Node sink) = forceLocal(Base::flow/2)(source, sink)
```

Comment:
```suggestion
    private predicate flowLocal(Node source, Node sink) = forceLocal(Base::flow/2)(source, sink)
```

Reply:
You made both local versions private - I only meant for the binary version to be private. I think it makes sense to expose the unary `flowToLocal`, as that has a reasonable local semantics, whereas a the binary predicate really only makes sense as a global predicate.